tfx/extensions/experimental/kfp_compatibility/kfp_container_component.py

```diff
@@ -0,0 +1,107 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Functions for creating container components from kubeflow components."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import yaml
+from google.protobuf.json_format import ParseDict
+from tfx.components.base import base_component
+from tfx.dsl.component.experimental import executor_specs, placeholders, container_component
+from tfx.extensions.experimental.kfp_compatibility.proto import kfp_component_spec_pb2
+from tfx.types.experimental.simple_artifacts import File
+from typing import Any, Callable, Dict, Text
+
+def load_kfp_yaml_container_component(
+    component_yaml_path: Text
+) -> Callable[..., base_component.BaseComponent]:
+  """Creates a container-based component from a Kubeflow component spec.
+
+  Args:
+    component_yaml_path: local file path of a Kubeflow Pipelines component
+                         YAML file.
+
+  Returns:
+    Container component that can be instantiated in a TFX pipeline.
+
+  Example:
+    component = load_kfp_yaml_container_component(
+      "kfp_pipelines_root/components/datasets/Chicago_Taxi_Trips/component.yaml"
+    )
+  """
+  with open(component_yaml_path) as component_file:
```

Comment:
This only works for local files. Do we anticipate usage which would permit this file to be on a remote location?

Code in tfx.utils.io_utils can help for those cases.

@charlesccychen for comment.